src/core/session.js

```diff
@@ -162,9 +162,9 @@ export class Session {
     const frameTarget = element.getAttribute("data-turbo-frame")
     const frame = frameTarget == "_top" ?
       null :
-      document.getElementById(frameTarget) || findClosestRecursively(element, "turbo-frame:not([disabled])")
+      FrameElement.getElementById(frameTarget) || findClosestRecursively(element, "turbo-frame")
 
-    if (isUnsafe || isStream || frame instanceof FrameElement) {
+    if (isUnsafe || isStream || (frame && !frame.disabled)) {
```

Comment:
@sfnelson since `FrameElement.getElementById` accounts for `instanceof FrameElement`, does this property check account for the concerns you've shared in https://github.com/hotwired/turbo/pull/1136/files/b750a8f2e44ef3133baf6328e5187c80ae5cf2e1#r1496721094?

Reply:
For the sake of shrinking the diff, both of these lines could likely be reverted in part:

```suggestion
    if (isUnsafe || isStream || frame instanceof FrameElement) {
```

Reply:
Yeah, that looks fine.

Would it make sense to add a static helper for finding the enclosing (non-disabled) FrameElement? In that case the `findClosestRecursively` could be replaced by e.g. `FrameElement.findClosest(element)` and the check would become `if (isUnsafe || isStream || frame) {`.

Reply:
For the purposes of code reuse, I can see two additional utility methods that would be helpful:

- `FrameElement.findEnclosingFrame(element)` would walk up the tree to find the enclosing frame (if any)
- `FrameElement.findTargetFrame(element)` would expose `#findFrameElement` – given an element it would identify which frame should be targeted – for use with links/forms that target frames for navigation

I think these would bring general utility to downstream consumers of Turbo – both of these cases have come up in our use of Frames to implement modals.